openwisp_notifications/static/openwisp-notifications/js/preferences.js

```diff
@@ -0,0 +1,449 @@
+'use strict';
+
+if (typeof gettext === 'undefined') {
+    var gettext = function(word){ return word; };
+}
+
+function getAbsoluteUrl(url) {
+    return notificationApiHost.origin + url;
+}
+
+(function ($) {
+    let isUpdateInProgress = false;
+
+    $(document).ready(function () {
+        const userId = $('.settings-container').data('user-id');
+        fetchGlobalSettings(userId);
+    });
+
+    function fetchGlobalSettings(userId) {
+        $.getJSON(getAbsoluteUrl(`/api/v1/notifications/user/${userId}/preference/`), function (globalData) {
```

Comment:
If you use `$.ajax` here, then you use the `beforeSend` callback for showing overlay and `complete` callback for hiding it. 

https://github.com/openwisp/openwisp-controller/blob/8001c3b9fa7783b5e33bfb5057d2cf5f2ce5bf25/openwisp_controller/connection/static/connection/js/commands.js#L348-L355